src/Compiler/TaskTableView.cpp

```diff
@@ -47,11 +47,27 @@ TaskTableView::TaskTableView(TaskManager *tManager, QWidget *parent)
 
 void TaskTableView::mousePressEvent(QMouseEvent *event) {
   auto idx = indexAt(event->pos());
-  bool expandAreaClicked = expandArea(idx).contains(event->pos());
-  if (expandAreaClicked) {
-    model()->setData(idx, expandAreaClicked, ExpandAreaRole);
+  auto task = m_taskManager->task(idx.row());
+  bool handled = false;
+
+  if (task && idx.data(Qt::DisplayRole).toString() == "Edit settings") {
+    auto parent = task->parentTask();
+    if (parent != nullptr) {
+      // Handle the event first so the selection moves before the modal dialog
+      QTableView::mousePressEvent(event);
+      handled = true;
```

Comment:
instead of this flag you can just return after emit.

Reply:
Sounds good, I try to avoid mid function returns out of habit, but if that's standard in our project I'll switch over to that.

Reply:
Not sure about standard but we have many returns in the middle of the function